### PR TITLE
[release-1.23] Bump kubernetes to 1.23.7

### PR DESCRIPTION
**Pull request by ncopa**
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md#changelog-since-v1236

Signed-off-by: Natanael Copa <ncopa@mirantis.com>

## Description

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ Though the Quick Start material is written for Debian/Ubuntu, you can use it for
     ```shell
     $ sudo k0s kubectl get nodes
     NAME   STATUS   ROLES    AGE    VERSION
-    k0s    Ready    <none>   4m6s   v1.23.6-k0s1
+    k0s    Ready    <none>   4m6s   v1.23.7-k0s1
```
Comment (twz123) [inline docs/install.md:67]: v1.23.7+k0s

File: docs/k0s-multi-node.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ sudo k0s kubectl get nodes
 
 ```shell
 NAME   STATUS   ROLES    AGE    VERSION
-k0s    Ready    <none>   4m6s   v1.23.6-k0s1
+k0s    Ready    <none>   4m6s   v1.23.7-k0s1
```
Comment (twz123) [inline docs/k0s-multi-node.md:147]: v1.23.7+k0s

**Review by twz123 — APPROVED** [review]
We also need to bump the versions in the docs of the main branch after the release has been cut.

File: docs/install.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ Though the Quick Start material is written for Debian/Ubuntu, you can use it for
     ```shell
     $ sudo k0s kubectl get nodes
     NAME   STATUS   ROLES    AGE    VERSION
-    k0s    Ready    <none>   4m6s   v1.23.6-k0s1
+    k0s    Ready    <none>   4m6s   v1.23.7-k0s
     ```
```
Comment (twz123) [inline docs/install.md:68]: v1.23.7+k0s

File: docs/k0s-multi-node.md [hunk 4]
```diff
@@ -144,7 +144,7 @@ sudo k0s kubectl get nodes
 
 ```shell
 NAME   STATUS   ROLES    AGE    VERSION
-k0s    Ready    <none>   4m6s   v1.23.6-k0s1
+k0s    Ready    <none>   4m6s   v1.23.7-k0s
```
Comment (twz123) [inline docs/k0s-multi-node.md:147]: v1.23.7+k0s

**Review by twz123 — APPROVED** [review]

**Review by trawler — APPROVED** [review]

**Pull request merged**